CMakeLists.txt

```diff
@@ -1088,15 +1088,15 @@ if(HPX_WITH_CUDA AND NOT HPX_WITH_CUDA_CLANG)
     hpx_library_dir(${CUDA_TOOLKIT_ROOT_DIR}/lib/x64)
     link_directories(${CUDA_TOOLKIT_ROOT_DIR}/lib/x64)
     set(CUDA_NVCC_FLAGS_DEBUG ${CUDA_NVCC_FLAGS_DEBUG};-D_DEBUG;-O0;-g;-G;-Xcompiler=-MDd;-Xcompiler=-Od;-Xcompiler=-Zi;-Xcompiler=-bigobj)
-    set(CUDA_NVCC_FLAGS_RELWITHDEBINFO ${CUDA_NVCC_FLAGS_RELWITHDEBINFO};-DNDEBUG;-O2;-g;-Xcompiler=-MD,-O2,-Zi;-Xcompiler=-bigobj)
+    set(CUDA_NVCC_FLAGS_RELWITHDEBINFO ${CUDA_NVCC_FLAGS_RELWITHDEBINFO};-DNDEBUG;-O2;-g;--verbose;-Xcompiler=-MD,-O2,-Zi;-Xcompiler=-bigobj;--keep)
```

Comment:
Sure. You have 'solved' the very same issue by unconditionally setting the compilation flags for all build configurations (here: https://github.com/STEllAR-GROUP/hpx/blob/3f87b2da11b02ff327d234adf5634d730a695bda/cmake/HPX_AddCompileFlag.cmake#L34). This however does not work for multi-target build systems like VS. So while I would like to solve it, I have no idea how to approach this issue as it seems to be a problem in FindCUDA.cmake not picking up settings from cmake generator constructs.